src/platform/Linux/bluez/Helper.cpp

```diff
@@ -1197,20 +1197,21 @@ static void UpdateAdditionalDataCharacteristic(BluezGattCharacteristic1 * charac
     CHIP_ERROR err = CHIP_NO_ERROR;
     chip::System::PacketBufferHandle bufferHandle;
 
-    char serialNumber[ConfigurationManager::kMaxSerialNumberLength + 1];
+    uint8_t rotatingDeviceIdUniqueId[ConfigurationManager::kRotatingDeviceIDUniqueIDLength];
```

Comment:
variable definitions should also be behind rotating device id ifdef - if you do not enable the rotating id, you should not use the stack space.